topics/galaxy-interface/tutorials/workflow-posters/tutorial.md

```diff
@@ -0,0 +1,271 @@
+> 1. **Opening the Workflow Preview**
+>    - Navigate to your Workflows list
+>    - Press "{% icon galaxy-eye %} View" on your desired Workflow
+>    - You should now see the preview page, which should look similar to this:
+>      ![Galaxys Workflow Preview page with the workflow being rendered next to an info-box](./images/workflow-preview.png)
+> 2. **Add the following to the URL in your browser:**
+>
+>    ```
+>    &embed=true&buttons=false&about=false&heading=false&minimap=false&zoom_controls=false
+>    ```
+> 
+> 3. **You should now see your Workflow rendered as shown below:**
+>
+>    ![Workflow preview page with all UI elements except for the Workflow itself hidden](./images/workflow-preview-embedded.png)
+>    In this view, you can still zoom in and out using your mouse-wheel, and drag the canvas around.
+>
+{: .hands_on}
+
+> <details-title>What does the text in the URL do?</details-title>
+>
+> These are so called query-parameters. They are chained to the existing parameters with a `&` symbol. 
+> Each parameter tells galaxy to render the UI slightly differently.
+> If you're curios at what they all do, try removing them individually and reloading the page.
+>
+{: .details}
+
+> <comment-title></comment-title>
+> If you cannot see a preview page like in the first screenshot, or if the query parameters have no effect,
+> you are on a Galaxy version older than 23.2
+>
+> You can still follow the rest of this tutorial to learn how to make high-resolution screenshots.
+{: .comment}
+
+
+## Exporting high-res screenshots
+
+Now that we have a view of our Workflow without the Galaxy UI, let's make a high-res screenshot using Firefox.
+
+<div class="Print" markdown="1">
+Firstly position your Workflow, so all of it is visible in the way you would like it to be on the poster.
+</div>
+
+<div class="Web" markdown="1">
+Firstly position your Workflow, so all of it is visible in the way you would like it to be on the screenshot.
+</div>
+
+> <tip-title>Can't fit your entire Workflow at once?</tip-title>
+> 
+> If you zoomed out as far as possible, but can't view all of your Workflow, try the following:
+> * Navigate back to the full Workflow preview page (without the UI hidden)
+> * Click anywhere outside of the Workflow preview (e.g. on the about section)
+> * Zoom out the entire browser window a bit, by using <kbd>Ctrl</kbd> + <kbd>Mouse Wheel</kbd>, or <kbd>Ctrl</kbd> + <kbd>-</kbd>
+> * Re-add the query parameters we added in the previous section
+> * If you still can't fit all of it, repeat the above steps
+{: .tip}
+
+> <hands-on-title>Creating a high-resolution screenshot</hands-on-title>
+> 
+> 1. **Open the Firefox Javascript console.** To do this, either:
+>    - Use the shortcut <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>K</kbd> (On OS X <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>K</kbd>)
+>    - Or navigate to the console using your mouse:
+>       * Menu
+>       * More Tools
+>       * Web Developer Tools
+>       * Console Tab
+>
```

Comment:
Oh, this is, hmm. _It makes sense_. I wish the screenshot tool exposed DPR, rather than just `1`, that would be an improvement they could make.

That said, so many websites have a lil `console.err` of "if someone told you to paste code in here, don't do it" so it feels like a maybe sub-optimal thing to teach end-users to do? (alternatively it's good that they learn how the web works, that console exists, but I don't think that's a step many want to add to their poster printing journey)

Should we just stick (whatever the current equivalent of) phantomjs is, in a Galaxy tool? And then have the user provide the URL to their workflow? We could automatically add the required URL parameters / automatically check galaxy version / error if it was too low. It would be a bit clunky, having to run a tool rather, but maybe it's something worth considering? (Though it would rely on the workflow contents being completely included in the preview, i don't know if we have a 'resize viewport to include all workflow steps' button/feature yet, so, that may be sub-optimal.)

We could then also run imagemagick in that tool (or separately), wouldn't be much work to add the palette options to https://github.com/bgruening/galaxytools/tree/master/tools/image_processing/graphicsmagick

Reply:
This discussion replaced by https://github.com/galaxyproject/galaxy/issues/17127